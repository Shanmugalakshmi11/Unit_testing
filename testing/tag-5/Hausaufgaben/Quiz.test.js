const quiz = require("./Quiz");

describe("isRightSolutionForQuestion function", () => {
  test("should return true for correct answer", () => {
    const questionKey = "frage_1";
    const correctAnswer = "Weil er immer kalt gelÃ¶tet ist.";

    const result = quiz.isRightSolutionForQuestion(questionKey, correctAnswer);
    console.log("result", result);
    expect(result).toBe(true);
  });

  test("should return false for incorrect answer", () => {
    const questionKey = "frage_2";
    const incorrectAnswer = "Because programmers love haircuts.";

    const result = quiz.isRightSolutionForQuestion(
      questionKey,
      incorrectAnswer
    );
    console.log("result", result);
    expect(result).toBe(false);
  });

  test("should throw an error for non-existing question", () => {
    const nonExistingQuestionKey = "non_existing_key";
    const answer = "Some answer";

    expect(() =>
      quiz.isRightSolutionForQuestion(nonExistingQuestionKey, answer)
    ).toThrow("No such question");
  });

  test("should throw an error for non-string answer", () => {
    const questionKey = "frage_3";
    const nonStringAnswer = 123;

    expect(() =>
      quiz.isRightSolutionForQuestion(questionKey, nonStringAnswer)
    ).toThrow("Answer can only be a string");
  });
});
