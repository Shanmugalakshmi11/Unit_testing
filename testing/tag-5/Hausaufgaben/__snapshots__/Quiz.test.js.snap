// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isRightSolutionForQuestion function should throw an error for non-existing question: No such question 1`] = `"isRightSolutionForQuestion is not a function"`;

exports[`isRightSolutionForQuestion function should throw an error for non-string answer: Answer can only be a string 1`] = `"isRightSolutionForQuestion is not a function"`;
